### ISSUE TITLE
'DataFrame' object has no attribute 'to_frame' with fastai 

### ISSUE THREAD

**Issue by ming-cui** [ModelOriented/DALEX#523]
I'm trying to wrap a `fastai` tabular `learner` with DALEX. I got `'DataFrame' object has no attribute 'to_frame'` error with `dx.Explainer(learn, xs, y, label = "Deep NN")`. Any potential problems with this line of code? Thanks!

**Comment by hbaniecki** [comment]
What is the verbose output from the Explainer?

[![image](https://user-images.githubusercontent.com/32574004/182252846-a8848177-a249-4152-9435-6c8366efaa49.png)](https://dalex.drwhy.ai/python-dalex-multioutput.html)

**Comment by ming-cui** [comment]
Please see the screenshot.
![Snipaste_2022-08-02_10-21-26](https://user-images.githubusercontent.com/49630086/182277869-6ef770f2-a268-4009-9b1d-f0737ee3e4bc.png)

**Comment by ming-cui** [comment]
Thanks, Hubert. fastai [`.predict`](https://docs.fast.ai/tabular.learner.html) method returns three objects. The source code is
![Snipaste_2022-08-02_21-11-34](https://user-images.githubusercontent.com/49630086/182382995-873d8f1e-ada8-44ab-822e-5af764be9ef5.png) Is it not compatible with DALEX? I'm still getting errors, and the messages were lengthy.

**Comment by hbaniecki** [comment]
Yes, as I said, you need to create a new function, which will use the `predict` method and return only one object - a 1-dimensional numpy array with predictions for a class of interest. Then pass it to the `predict_function` parameter in `dx.Explainer`

**Comment by ming-cui** [comment]
Hi Hubert, I think I got the basic idea, but I struggled for quite a while as I'm new to Python and fastai. I got the error as illustrated in the screenshot. It seems the `.predict` function only takes a row as input. I guess I need to write some customized prediction function to pass into DALEX. I've linked the source code of `.predict` [here](https://github.com/fastai/fastai/blob/master/nbs/43_tabular.learner.ipynb), or you can view the code in the sceenshot. Could you help with getting the right prediction function to work with DALEX? Thank you so much!
![Error prediction function](https://user-images.githubusercontent.com/49630086/182606987-7d26a31a-8e0b-4090-9384-4c652f5beb10.png)

![.predict source code](https://user-images.githubusercontent.com/49630086/182607945-fb275687-fbb2-4504-a898-4175a06f87d2.png)

**Comment by hbaniecki** [comment]
Hi! Maybe you can try `predict_function = lambda m, d: m.predict(d)[2]`?

**Comment by ming-cui** [comment]
Thanks for the function. I got the error message `'DataFrame' object has no attribute 'to_frame'` with details below. Will the `.get_preds()` method be useful (shown in the photo in my last reply)? `.get_preds()` returns tuples (one for predictions and the other for true labels), and I haven't figured out how to put it in a `explainer`.
![Snipaste_2022-08-03_20-59-07](https://user-images.githubusercontent.com/49630086/182614611-c886d039-0665-430e-a55e-d64b1916cf26.png)
.

**Comment by ming-cui** [comment]
Is it possible that the tensor format predictions returned by `.predict()` caused `'DataFrame' object has no attribute 'to_frame'`?

**Comment by hbaniecki** [comment]
Yes, you need to first apply the predict_function outside of the explainer and see what it returns.

**Comment by ming-cui** [comment]
It returns a tensor. For example, `learn.predict(xs_nn.iloc[0])[2]` will return `tensor([1.7050])` as the predicted value based on the first row of the training dataframe `xs_nn`. What should I do with that? Many thanks!

**Comment by hbaniecki** [comment]
It should return a one dimensional numpy array
